interface/login/login.php

```diff
@@ -160,19 +160,23 @@ function transmit_form(element) {
             element.disabled = true;
             // nothing fancy. mainly for mobile.
             element.innerHTML = '<i class="fa fa-sync fa-spin"></i> ' + jsText(<?php echo xlj("Authenticating"); ?>);
-            <?php if (!empty($GLOBALS['restore_sessions'])) { ?>
+            <?php if (session_name()) { ?>
+                <?php $scparams = session_get_cookie_params(); ?>
                 // Delete the session cookie by setting its expiration date in the past.
                 // This forces the server to create a new session ID.
                 var olddate = new Date();
                 olddate.setFullYear(olddate.getFullYear() - 1);
-                <?php if (version_compare(phpversion(), '7.3.0', '>=')) { ?>
-                    // Using the SameSite setting when using php version 7.3.0 or above, which avoids browser warnings when cookie is not 'secure' and SameSite is not set to anything
-                    document.cookie = <?php echo json_encode(urlencode(session_name())); ?> + '=' + <?php echo json_encode(urlencode(session_id())); ?> + '; path=<?php echo($web_root ? $web_root : '/');?>; expires=' + olddate.toGMTString() + '; SameSite=Strict';
-                <?php } else { ?>
-                    document.cookie = <?php echo json_encode(urlencode(session_name())); ?> + '=' + <?php echo json_encode(urlencode(session_id())); ?> + '; path=<?php echo($web_root ? $web_root : '/');?>; expires=' + olddate.toGMTString();
```

Comment:
hi @sunsetsystems , just realized why had the urlencode here, which documented here (as I recall, when a comma is in there, it doesn't work as it should):
https://github.com/openemr/openemr/blob/master/src/Common/Session/SessionUtil.php#L25-L27

just checking to make sure you are ok with me adding that back?

Reply:
I'm not seeing why a comma would mess it up... special characters with meaning in cookies are semicolon, equals sign and space. Also not sure that urlencode is the right method of encoding?

Reply:
looked back into this. Looks like php does this without telling us:
![image](https://user-images.githubusercontent.com/278968/117257269-8f291780-ae00-11eb-97d4-bfd8a77f6981.png)
(note comma is set as %2C by session_start())
So, when we manually add a session id with a comma, then need to encode it or it won't match the one set by php

Reply:
also to add, urlencode just happens to convert that comma to %2C :)

Reply:
Bottom line, I'm confused about this. Can't find an authoritative source that says what encoding, if any, PHP expects with session IDs in cookies.

Reply:
yep, this was a huge painful bug to sort out (I am beginning to remember the pain) when we went to sid_bits_per_character of 6 to increase the randomness of the session id. The bug is more obvious if your remove the urlencode's from the restoresession script. Then the session separations essentially collapse, but only when there is a comma in the session (happens about 1 in 3 session id's). The documentation is super poor on this and I recall finally finding something that discussed this on the internet and noted that PHP was encoding the comma in the cookie it produced. fun stuff (to convince yourself rec. removing the urlencode's on the restoresession script) :)

Reply:
Alrighty then, please put them back! Thanks!

Reply:
oh goodness, just tested after removed urlencode's from the restoresession script. Log in and out until you get a cookie with a %2C in it, then watch the magic :)
First the restoresession will convert the cookie id %2C to a actual comma, then it will change it to a new session id on next click, and will then get the infamous "Site ID is missing from session data!" message :)

Reply:
urlencode it is then. :)

Reply:
done :)